prow/cmd/branchprotector/protect_test.go

```diff
@@ -631,51 +722,46 @@ func TestProtect(t *testing.T) {
 			}
 		}
 
+		var cfg config.Config
+		err := yaml.Unmarshal([]byte(tc.config), &cfg)
```

Comment:
What's the benefit of pulling in yaml marshal into the unit under test here?

Reply:
Easier for humans to read and write, also better test case (since what we really care about is ensuring that yaml string foo results in X protection)